ComputerVision.cs

```diff
@@ -0,0 +1,518 @@
+    [Tooltip("Wether to show the confidence level next to the detected object or not, ex.(car 100%)")]
+    public bool showConfidenceLevel = true;
+
+    [Tooltip("For object detection, the font of the object class font")]
+    public int objectClassFontSize = 30;
+
+    [Tooltip("For object detection, color of the object class font if the confidence level is above or equal the confidence threshold")]
+    public Color highConfidenceFontColor = Color.green;
+
+    [Tooltip("For object detection, color of the object class font if the confidence level is below the confidence threshold")]
+    public Color lowConfidenceFontColor = Color.red;
+
+    public int confidenceThreshold = 1;
+
+    public enum ConnectionMode
+    {
+        Rest,
+        WebSocket
+    }
+
+    public enum ServiceMode
+    {
+        FaceDetection,
+        ObjectDetection
+    }
+
+    MobiledgeXWebSocketClient client;
+    MobiledgeXIntegration integration;
+    /// <summary>
+    /// Hide Detection rects before taking a screenshot
+    /// </summary>
+    static bool showGUI;
+    int[][] faceDetectionRects;
+    static @Object[] objectsDetected;
+
+
+    bool serviceAlreadyStarted
+    {
+        get
+        {
+            switch (serviceMode)
+            {
+                case ServiceMode.ObjectDetection:
+                    return objectsDetected == null ? false : true;
+                case ServiceMode.FaceDetection:
+                    return faceDetectionRects == null ? false : true;
+                default:
+                    return false;
+            }
+        }
+    }
+
+    /// <summary>
+    /// For more info about the computer vision server and the url suffix check MobiledgeX docs
+    /// for Rest https://developers.mobiledgex.com/guides-and-tutorials/computer-vision/how-to-computer-vision-api
+    /// for WebSockets https://mobiledgex.com/blog/2020/04/22/computer-vision-websocket-support
+    /// </summary>
+    string urlSuffix
+    {
+        get
+        {
+            switch (serviceMode)
+            {
+                case ServiceMode.ObjectDetection:
+                    return "/object/detect/";
+                case ServiceMode.FaceDetection:
+                    return "/detector/detect/";
+                default:
+                    return "";
+            }
+        }
+    }
+
+    /// <summary>
+    /// The complete url retreived from FindCloudlet+ url suffix
+    /// </summary>
+    static string url;
+    static bool urlFound; // trigger indicating RegisterAndFindCloudlet, GetAppPort() & GetUrl() occured
+    bool webRequestsLock = false; // to control the flow of sending web requests
+    bool wsStarted; // trigger indicating wether websocket connection have been intialized or not
+
+
+
+    #region Monobehaviour callbacks
+    IEnumerator Start()
+    {
+#if UNITY_EDITOR
+        GetEdgeConnection();
+        yield break;
+#endif
+        if (Application.internetReachability == NetworkReachability.NotReachable)
```

Comment:
hmm. Is this a Unity property? Why this over HasCellular check or IsEdgeEnabled?

Reply:
Yes, wonder what the test is for this.

Reply:
It's  a simple check for Internet Connection
https://docs.unity3d.com/ScriptReference/NetworkReachability.html